### PR TITLE
Update android gradle plugin 2.2.0

### PR DESCRIPTION
**Pull request by jaredsburrows**
- Use the plugins repo to mirror mavencentral and jcenter
- Updated android gradle plugin to 2.2.0 (is on jcenter())
- Update to gradle 3.1

### CODE REVIEW

File: butterknife-integration-test/build.gradle [hunk 1]
```diff
@@ -37,7 +36,7 @@ android {
 
 dependencies {
   compile project(':butterknife')
-  apt project(':butterknife-compiler')
+  annotationProcessor project(':butterknife-compiler')
```
Comment (jaredsburrows) [inline butterknife-integration-test/build.gradle]: Remote `apt` in favor of the annotation processor that comes with the plugin.

File: butterknife-integration-test/src/test/java/com/example/butterknife/library/SimpleActivityTest.java [hunk 2]
```diff
@@ -6,11 +6,11 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.robolectric.Robolectric;
-import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RobolectricTestRunner;
 
 import static com.google.common.truth.Truth.assertThat;
 
-@RunWith(RobolectricGradleTestRunner.class)
+@RunWith(RobolectricTestRunner.class)
```
Comment (jaredsburrows) [inline butterknife-integration-test/src/test/java/com/example/butterknife/library/SimpleActivityTest.java]: Deprecation.

File: butterknife-integration-test/build.gradle (hunk 1 above)
Comment (jaredsburrows) [inline butterknife-integration-test/build.gradle]: Remote `apt` in favor of the annotation processor that comes with the plugin.

File: butterknife-integration-test/src/test/java/com/example/butterknife/library/SimpleAdapterTest.java [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
 import static com.google.common.truth.Truth.assertThat;
 
-@RunWith(RobolectricGradleTestRunner.class)
+@RunWith(RobolectricTestRunner.class)
 public class SimpleAdapterTest {
   @Test public void verifyViewHolderViews() {
     Context context = RuntimeEnvironment.application;
 
-    View root = LayoutInflater.from(context).inflate(R.layout.simple_list_item, null);
+    View root = View.inflate(context, R.layout.simple_list_item, null);
```
Comment (jaredsburrows) [inline butterknife-integration-test/src/test/java/com/example/butterknife/library/SimpleAdapterTest.java]: Fix lint warning.

File: butterknife/build.gradle [hunk 4]
```diff
@@ -33,9 +33,23 @@ dependencies {
   testCompile deps.junit
   testCompile deps.truth
   testCompile deps.compiletesting
-  testCompile files(org.gradle.internal.jvm.Jvm.current().getRuntimeJar())
+  testCompile files(getRuntimeJar())
   testCompile files(org.gradle.internal.jvm.Jvm.current().getToolsJar())
   testCompile project(':butterknife-compiler')
 }
 
+def getRuntimeJar() {
```
Comment (jaredsburrows) [inline butterknife/build.gradle]: @JakeWharton `org.gradle.internal.jvm.Jvm.current().getRuntimeJar()` was removed. See here: https://groups.google.com/forum/#!topic/sonarqube/DnE1npMpaiA and https://github.com/bfarka/sonar-gradle/blob/295c365f459cd19617b4fe8ce62bdb58b999bd74/src/main/java/org/sonarqube/gradle/SonarQubePlugin.java#L370.

File: sample/library/build.gradle [hunk 5]
```diff
@@ -1,11 +1,10 @@
 buildscript {
   dependencies {
-    classpath 'com.jakewharton:butterknife-gradle-plugin:8.2.0'
+    classpath 'com.jakewharton:butterknife-gradle-plugin:8.4.0'
```
Comment (jaredsburrows) [inline sample/library/build.gradle]: Left over from previous PR.

File: build.gradle [hunk 6]
```diff
@@ -3,7 +3,9 @@ subprojects { project ->
   version = VERSION_NAME
 
   repositories {
-    mavenCentral()
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
```
Comment (JakeWharton) [inline build.gradle]: why?

Comment (jaredsburrows) [inline build.gradle]: @JakeWharton you were already using it here(https://github.com/JakeWharton/butterknife/pull/749/files#diff-c197962302397baf3a4cc36463dce5eaR34). This mirrors both `mavenCentral()` and `jcenter()`.

Comment (jaredsburrows) [inline build.gradle]: @JakeWharton Is this good?

**Pull request merged**